packages/ui/src/components/filters/filters-bar/actions/views.tsx

```diff
@@ -0,0 +1,241 @@
+      return
+    }
+
+    saveView(newViewName)
+    setNewViewName('')
+    setNameError('')
+    setIsNewViewDialogOpen(false)
+  }
+
+  /**
+   * Handles renaming existing view
+   * - Validates current view exists
+   * - Checks for duplicate names
+   * - Updates view name and closes dialog
+   */
+  const handleRename = () => {
+    if (!currentView) return
+
+    if (checkNameExists(newViewName, currentView.id)) {
+      setNameError('A view with this name already exists.')
+      return
+    }
+
+    renameView(currentView.id, newViewName)
+    setNewViewName('')
+    setNameError('')
+    setIsRenameDialogOpen(false)
+  }
+
+  // Validation flags for enabling/disabling save buttons
+  const canSaveNew = newViewName.trim() !== ''
+  const canRename = currentView && newViewName.trim() !== '' && newViewName !== currentView.name
+
+  return (
+    <>
+      {!currentView ? (
+        <Button
+          className="flex items-center gap-x-1.5 text-14 text-foreground-4 hover:text-foreground-1 transition-colors duration-200 px-0"
+          variant="custom"
+          onClick={() => setIsNewViewDialogOpen(true)}
+        >
+          <Icon name="bookmark-add" size={12} />
+          <span>Save view</span>
+        </Button>
+      ) : (
+        <DropdownMenu>
+          <DropdownMenuTrigger asChild>
+            <Button
+              className="flex items-center gap-x-1.5 text-14 text-foreground-4 hover:text-foreground-1 transition-colors duration-200 px-0"
+              variant="custom"
+            >
+              <Icon name="bookmark-icon" size={12} />
+              <span>Manage view</span>
+            </Button>
+          </DropdownMenuTrigger>
+          <DropdownMenuContent className="w-48" align="end">
+            {hasChanges && (
+              <>
+                <DropdownMenuItem onSelect={() => updateView(currentView)}>Update current view</DropdownMenuItem>
+                <DropdownMenuItem onSelect={() => setIsNewViewDialogOpen(true)}>Save as new view</DropdownMenuItem>
+              </>
+            )}
+            <DropdownMenuItem
+              onSelect={() => {
+                setNewViewName(currentView.name)
+                setIsRenameDialogOpen(true)
+              }}
+            >
+              Rename
+            </DropdownMenuItem>
+            <DropdownMenuItem className="text-foreground-danger" onSelect={() => deleteView(currentView.id)}>
+              Delete view
+            </DropdownMenuItem>
+          </DropdownMenuContent>
+        </DropdownMenu>
+      )}
+
+      <AlertDialog open={isNewViewDialogOpen} onOpenChange={setIsNewViewDialogOpen}>
+        <AlertDialogContent className="max-w-[460px]">
+          <AlertDialogHeader>
+            <AlertDialogTitle>New view</AlertDialogTitle>
+          </AlertDialogHeader>
+          <AlertDialogDescription variant="secondary">
+            You can save current configuration of the view. Manage and switch them through the{' '}
```

Comment:
can we externalise all display strings to appropriate language files in the first cut itself? it'll reduce re-work to add i18n later.

Reply:
I’m planning to address this in the next PR so that this one doesn’t block further development.